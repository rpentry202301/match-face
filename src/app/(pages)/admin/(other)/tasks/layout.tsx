export const metadata = {
  title: {
    default: "タスク一覧",
    // Todo: 正式なアプリ名に差し替え
    template: `%s | 管理者画面 | Match Face`,
  },
  description: "管理者質問・解答の追加/編集ページ",
};

export default function TasksLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
