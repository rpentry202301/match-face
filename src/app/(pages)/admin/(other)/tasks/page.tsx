import TasksIndex from "@/components/pages/admin/tasks/index/TasksIndex";

/**
 * @author Hayato Kobayashi
 */
const TasksPage = ({ searchParams }: Props) => {
  return <TasksIndex searchParams={searchParams} />;
};

type Props = {
  searchParams: {
    departmentId: string | string[] | undefined;
    searchKeyword: string | string[] | undefined;
  };
};

export default TasksPage;
