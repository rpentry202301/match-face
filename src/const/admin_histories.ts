export const skills = [
    {id:1,name:"JavaScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:"TypeScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:"React",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:"Next.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:"Vue.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,name:"Angular",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:7,name:"Nuxt.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:8,name:"jQuery",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,name:"Redux",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:10,name:"vuex",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:11,name:"MaterialUI",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:12,name:"tailwind",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:13,name:"Bootstrap",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:14,name:"Vercel",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const departments = [
    {id:1,name:"FR",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:2,name:"Java",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:3,name:"PHP",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:4,name:"CL",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:5,name:"ML",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:6,name:"QA",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
]

export const answer_request_groups =[
    {id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
]
