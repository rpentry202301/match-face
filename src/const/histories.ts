export type DataType = {
  id: number;
  comment_created_at: boolean;
  name: string;
  detail: string;
  answer_update_at: string;
  skill_id: number[];
}[];

export type SkillType = {
  id: number;
  name: string;
  createdUser: string;
  createdAt: string;
  updateUser: string;
  updateAt: string;
}[];

export const projects: DataType = [
  {
    id: 1,
    comment_created_at: true,
    name: "ポムポムプリン",
    detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [1, 2, 3],
  },
  {
    id: 2,
    comment_created_at: false,
    name: "日暮らし、",
    detail:
      "硯にむかひて、心にうつりゆくよしなし事を、そこはかとなく書きつくれば、あやしうこそものぐるほしけれ。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [2, 3, 4],
  },
  {
    id: 3,
    comment_created_at: false,
    name: "親譲りの無鉄砲で",
    detail:
      "小供の時から損ばかりしている。小学校に居る時分学校の二階から飛び降りて一週間ほど腰を抜かした事がある。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [5, 6, 7],
  },
  {
    id: 4,
    comment_created_at: true,
    name: "なぜそんな無闇を",
    detail:
      "したと聞く人があるかも知れぬ。別段深い理由でもない。新築の二階から首を出していたら、同級生の一人が冗談に、いくら威張っても、そこから飛び降りる事は出来まい。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [1, 2, 6, 7],
  },
  {
    id: 5,
    comment_created_at: true,
    name: "ポムポムプリン",
    detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_update_at: "2023-10-10T16:52:46.053Z",
    skill_id: [3, 4, 5],
  },
  {
    id: 6,
    comment_created_at: false,
    name: "弱虫やーい。",
    detail:
      "と囃したからである。小使に負ぶさって帰って来た時、おやじが大きな眼をして二階ぐらいから飛び降りて腰を抜かす奴があるかと云ったから",
    answer_update_at: "2023-10-10T16:52:46.053Z",
    skill_id: [1, 2, 3],
  },
  {
    id: 7,
    comment_created_at: true,
    name: "つれづれなるまゝ",
    detail:
      "日暮らし、硯にむかひて、心にうつりゆくよしなし事を、そこはかとなく書きつくれば、あやしうこそものぐるほしけれ。",
    answer_update_at: "2023-10-10T16:52:46.053Z",
    skill_id: [1, 2, 3],
  },
];

// export {data};

export const skills = [
  {
    id: 1,
    skill: "JavaScript",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 2,
    skill: "TypeScript",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 3,
    skill: "React",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 4,
    skill: "Next.js",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 5,
    skill: "Vue.js",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 6,
    skill: "Angular",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 7,
    skill: "Nuxt.js",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 8,
    skill: "jQuery",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 9,
    skill: "Redux",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 10,
    skill: "vuex",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 11,
    skill: "MaterialUI",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 12,
    skill: "tailwind",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 13,
    skill: "Bootstrap",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 14,
    skill: "Vercel",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
];
