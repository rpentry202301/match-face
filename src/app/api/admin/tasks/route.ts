import { Answer_RequestsType } from "@/types/admin/tasks/types";
import { NextResponse } from "next/server";

/**
 * @author Hayato Kobayashi
 * @param req Next API request
 * @returns Next API responce
 *
 * @todo_1 検索ワード(searchKeyword), 所属(departmentId)で検索条件を指定
 */
export async function GET(req: Request) {
  // todo_1
  const res = await fetch(`${process.env.BE_URL}/answer_requests`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
    },
  });
  // console.log("res", res.status);
  const answerRequests: Answer_RequestsType = await res.json();
  const tasks = answerRequests.answerRequests; // fetch時に"answerRequests.answerRequestsの形になるので修正"
  return NextResponse.json(tasks);
}
