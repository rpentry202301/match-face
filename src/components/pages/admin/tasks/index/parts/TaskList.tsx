import { TasksType } from "@/types/admin/tasks/types";

/**
 * @author Hayato Kobayashi
 * @todo_1 レスポンシブ対応
 */
const TaskList = ({ tasks }: { tasks: TasksType[] }) => {
  if (tasks.length === 0) return <div>表示するデータがありません</div>;

  return (
    <div data-testid="data-table">
      <table className="border-collapse border-2 border-deep-gray text-sm">
        <thead className="bg-light-gray">
          <tr>
            <th className="border-2 border-deep-gray p-2 text-center w-72">
              案件名
            </th>
            <th className="border-2 border-deep-gray p-2 text-center w-16">
              質問数
            </th>
            <th className="border-2 border-deep-gray p-2 text-center w-16">
              職種
            </th>
            <th className="border-2 border-deep-gray p-2 text-center w-72">
              回答ユーザー
            </th>
            <th className="border-2 border-deep-gray p-2 text-center w-28">
              期日
            </th>
          </tr>
        </thead>
        <tbody>
          {tasks.map((task) => (
            <tr key={task.id}>
              <td className="border-2 border-deep-gray p-2 text-center">
                {task.project.name}
              </td>
              <td className="border-2 border-deep-gray p-2 text-center">
                {task.questionCount}
              </td>
              <td
                className="border-2 border-deep-gray p-2 text-center"
                data-testid={`task-data-${task.department}`}
              >
                {task.department.name}
              </td>
              <td className="border-2 border-deep-gray p-2 text-center">
                {/* 回答ユーザーは3名まで表示。それ以降は"..."で省略 */}
                {task.answerUserList.length > 3
                  ? task.answerUserList // ex)テスト太郎、テスト次郎、テスト三郎...
                      .slice(0, 3)
                      .map((user, i) =>
                        i !== 3 ? (
                          <span key={user.userId}>{user.userName}、</span>
                        ) : (
                          <span key={user.userId}>{user.userName}...</span>
                        )
                      )
                  : task.answerUserList.map(
                      (
                        user,
                        i // ex)テスト太郎、テスト次郎、テスト三郎
                      ) => (
                        <span key={user.userId}>
                          {user.userName}
                          {i + 1 !== task.answerUserList.length && "、"}
                        </span>
                      )
                    )}
              </td>
              <td className="border-2 border-deep-gray p-2 text-center">
                {task.deadline.toString().slice(0, 10)}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default TaskList;
