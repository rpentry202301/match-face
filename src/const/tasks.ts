import { Answer_RequestsType } from "@/types/admin/tasks/types";

// テスト用データ
export const departments = [
  {
    id: 1,
    name: "Java",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 2,
    name: "PHP",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 3,
    name: "FR",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 4,
    name: "CL",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 5,
    name: "ML",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 6,
    name: "QA",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
];

export const answer_requests: Answer_RequestsType = {
  answerRequests: [
    {
      id: 1,
      project: {
        id: 1,
        name: "バックエンド案件",
        detail: "販促アプリの新規開発、既存システムの保守・運用。",
        enterpriseId: 0,
        departmentId: 0,
        questionList: [],
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
        deleted: false,
      },
      department: {
        id: 1,
        name: "Java",
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
      },
      answerUserList: [
        {
          answerRequestId: 1,
          userId: 1,
          userName: "テスト太郎",
          answered: false,
        },
      ],
      questionCount: 2,
      deadline: "2023-12-01T18:00:00",
      createdUser: "テスト用",
      createdAt: "2023-09-11T11:51:22.182407",
      updateUser: "テスト用",
      updateAt: "2023-09-11T11:51:22.182407",
    },
    {
      id: 2,
      project: {
        id: 2,
        name: "バックエンド案件",
        detail: "販促アプリの新規開発、既存システムの保守・運用。",
        enterpriseId: 0,
        departmentId: 0,
        questionList: [],
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
        deleted: false,
      },
      department: {
        id: 2,
        name: "PHP",
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
      },
      answerUserList: [
        {
          answerRequestId: 2,
          userId: 2,
          userName: "テスト太郎",
          answered: false,
        },
      ],
      questionCount: 2,
      deadline: "2023-12-01T18:00:00",
      createdUser: "テスト用",
      createdAt: "2023-09-11T11:51:22.182407",
      updateUser: "テスト用",
      updateAt: "2023-09-11T11:51:22.182407",
    },
    {
      id: 3,
      project: {
        id: 3,
        name: "フロントエンド案件",
        detail: "販促アプリの新規開発、既存システムの保守・運用。",
        enterpriseId: 0,
        departmentId: 0,
        questionList: [],
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
        deleted: false,
      },
      department: {
        id: 3,
        name: "FR",
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
      },
      answerUserList: [
        {
          answerRequestId: 3,
          userId: 3,
          userName: "テスト太郎",
          answered: false,
        },
      ],
      questionCount: 2,
      deadline: "2023-12-01T18:00:00",
      createdUser: "テスト用",
      createdAt: "2023-09-11T11:51:22.182407",
      updateUser: "テスト用",
      updateAt: "2023-09-11T11:51:22.182407",
    },
  ],
};
