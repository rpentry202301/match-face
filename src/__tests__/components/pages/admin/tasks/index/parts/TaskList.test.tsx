import TaskList from "@/components/pages/admin/tasks/index/parts/TaskList";
import { answer_requests } from "@/const/tasks";
import { cleanup, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("TaskList.tsx", () => {
  beforeAll(() => {
    jest.clearAllMocks();
    cleanup();
  });
  afterAll(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it("スナップショットテスト", () => {
    const { container } = render(
      <TaskList tasks={answer_requests.answerRequests} />
    );
    expect(container).toMatchSnapshot();
  });
  it("表示できるデータが存在しない", () => {
    cleanup();
    render(<TaskList tasks={[]} />);

    const noDatatxt = screen.getByText("表示するデータがありません");
    expect(noDatatxt).toBeInTheDocument();
    const dataTable = screen.queryByTestId("data-table");
    expect(dataTable).toBeNull();
  });
});
