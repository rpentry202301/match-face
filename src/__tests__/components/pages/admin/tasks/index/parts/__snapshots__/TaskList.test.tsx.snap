// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList.tsx スナップショットテスト 1`] = `
<div>
  <div
    data-testid="data-table"
  >
    <table
      class="border-collapse border-2 border-deep-gray text-sm"
    >
      <thead
        class="bg-light-gray"
      >
        <tr>
          <th
            class="border-2 border-deep-gray p-2 text-center w-72"
          >
            案件名
          </th>
          <th
            class="border-2 border-deep-gray p-2 text-center w-16"
          >
            質問数
          </th>
          <th
            class="border-2 border-deep-gray p-2 text-center w-16"
          >
            職種
          </th>
          <th
            class="border-2 border-deep-gray p-2 text-center w-72"
          >
            回答ユーザー
          </th>
          <th
            class="border-2 border-deep-gray p-2 text-center w-28"
          >
            期日
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            バックエンド案件
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
            data-testid="task-data-[object Object]"
          >
            Java
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            <span>
              テスト太郎
            </span>
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2023-12-01
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            バックエンド案件
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
            data-testid="task-data-[object Object]"
          >
            PHP
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            <span>
              テスト太郎
            </span>
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2023-12-01
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            フロントエンド案件
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
            data-testid="task-data-[object Object]"
          >
            FR
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            <span>
              テスト太郎
            </span>
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2023-12-01
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
